core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -447,4 +450,39 @@ public void unmountException() throws Exception {
       assertSame(EXCEPTION, e);
     }
   }
+
+  /**
+   * Ensures Warnings are logged or exceptions are thrown when an {@link AlluxioURI} with a
+   * scheme://host is passed.
+   */
+  @Test
+  public void UriCheckTest() throws Exception {
+    Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
+    Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    AlluxioURI uri = new AlluxioURI("alluxio://localhost:1234/root");
+    boolean exceptionRaised = true;
+    try {
+      BaseFileSystem.UriCheck(uri);
+      exceptionRaised = false;
+      fail("Should have failed on bad host and port");
+    } catch (IllegalArgumentException e) {
+      assertTrue(exceptionRaised);
+    }
+
+    uri = new AlluxioURI("alluxio://localhost:19998/root");
+    try {
+      BaseFileSystem.UriCheck(uri);
+    } catch (IllegalArgumentException e) {
```

Comment:
we usually wouldn't catch an unexpected exception like this. If any type of exception is thrown and not caught, the test will fail and report the exception.